ClassIsland/Views/SettingPages/WeatherSettingsPage.xaml.cs

```diff
@@ -53,9 +53,48 @@ private void ButtonEditCurrentCity_OnClick(object sender, RoutedEventArgs e)
         OpenDrawer("CitySearcher");
     }
 
+    /// <summary>
+    /// [搜索城市或地区] 防抖定时器初始化
+    /// </summary>
+    /// <param name="sender"></param>
+    /// <param name="e"></param>
+    private void WeatherSettingsPage_Loaded(object sender, RoutedEventArgs e)
+    {
+        // 初始化防抖定时器，设置间隔时间为50毫秒
+        SearchDebounceTimer = new DispatcherTimer { Interval = TimeSpan.FromMilliseconds(50) };
+        SearchDebounceTimer.Tick += SearchDebounceTimer_Tick;
+        SearchDebounceTimer.Stop();
+    }
+
+    /// <summary>
+    /// [搜索城市或地区] TextBox的文本改变事件
+    /// </summary>
+    /// <param name="sender"></param>
+    /// <param name="e"></param>
     private void TextBoxSearchCity_OnTextChanged(object sender, TextChangedEventArgs e)
     {
-        ViewModel.CitySearchResults = WeatherService.GetCitiesByName(((TextBox)sender).Text);
+        // 为全局变量赋值
+        GlobalTextBoxSearchCity = (TextBox)sender;
+
+        // 重置定时器
+        SearchDebounceTimer.Stop();
+        SearchDebounceTimer.Start();
+    }
+
+    /// <summary>
+    /// [搜索城市或地区] 防抖定时器触发事件
+    /// </summary>
+    /// <param name="sender"></param>
+    /// <param name="e"></param>
+    private void SearchDebounceTimer_Tick(object? sender, EventArgs e)
+    {
+        // 停止定时器
+        SearchDebounceTimer.Stop();
+
+        // 更新搜索结果
+        var searchText = GlobalTextBoxSearchCity.Text;
+        ViewModel.CitySearchResults =
+            Task.Run(() => WeatherService.GetCitiesByName(searchText)).Result;
```

Comment:
这里可以考虑下用异步方法来获取天气，这样在获取天气信息时不会阻塞 UI 线程，可以提升用户在搜索时的流畅度（特别是在网络不好的情况下）：

```suggestion
            await WeatherService.GetCitiesByName(searchText);
```